metrics-api/src/main/java/software/amazon/swage/metrics/record/MetricRecorder.java

```diff
@@ -221,4 +232,50 @@ protected abstract void record(
      */
     protected void close(R context) {
     }
+
+    private final class DefaultMetricContext implements MetricContext {
+        private final MetricRecorder recorder;
+        private final RecorderContext context;
+        private final MetricContext parent;
+
+        public DefaultMetricContext(MetricRecorder recorder, TypedMap attributes) {
+            this(recorder, recorder.newRecorderContext(attributes), null);
+        }
+
+        DefaultMetricContext(MetricRecorder recorder, RecorderContext context, MetricContext parent) {
+            this.recorder = recorder;
+            this.context = context;
+            this.parent = parent;
+        }
+
+        @Override
+        public TypedMap attributes() {
+            return context.attributes();
+        }
+
+        @Override
+        public void record(Metric label, Number value, Unit unit, Instant time) {
+            recorder.record(label, value, unit, time, context);
+        }
+
+        @Override
+        public void count(Metric label, long delta) {
+            recorder.count(label, delta, context);
+        }
+
+        @Override
+        public void close() {
+            recorder.close(context);
+        }
+
+        @Override
+        public MetricContext newChild(TypedMap attributes) {
+            return new DefaultMetricContext(recorder, context.newChild(attributes), this);
+        }
+
+        @Override
+        public MetricContext parent() {
+            return null;
```

Comment:
Am I missing something, or doesn't this break MetricContext#parent for all DefaultMetricContext instances?